File: CSharp/Mayope.Api/Requests/AuthenticatedRequest.cs
using System;
using System.ComponentModel.DataAnnotations;

using Newtonsoft.Json;

namespace Mayope.Api.Requests;
{
    /// <summary>
    /// A request including an authentication token.
    /// </summary>
    public abstract class AuthenticatedRequest : IRequest
    {
        /// <summary>
        /// A token authenticating the user.
        /// </summary>
        [Required]
        [JsonRequired]
        public Guid Token { get; set; }
    }
}

File: Java/org/mayope/api/requests/AuthenticatedRequest.java
package org.mayope.api.requests;

import java.util;

/**
 * A request including an authentication token.
 */
public abstract class AuthenticatedRequest implements Request {
    /**
     * A token authenticating the user.
     */
    private UUID token;

    /**
     * Get a token authenticating the user.
     *
     * @return The value
     */
    public UUID getToken() {
        return token;
    }

    /**
     * Set a token authenticating the user.
     *
     * @param token The new value
     */
    public void setToken(UUID token) {
        this.token = token;
    }
}